### PR TITLE
Issue #3232309 by navneet0693: Switched if-else condition in social_embed_post_update_11001_populate_field_embed_content_settings.

### PR DESCRIPTION
**Pull request by navneet0693**
## Problem
The if condition was meant for executing the update in database but it was in-correctly written. So, if the conditions were true, it said Batch API that it has finished instead of doing the update. 

## Solution
Switch the if-else condition logic.

## Issue tracker
https://www.drupal.org/project/social/issues/3232309

## Theme issue tracker
N.A

## How to test
- [ ] Checkout to this branch
- [ ] Do a `drush updb -y`
- [ ] Check the values in table `user__field_user_embed_content_consent` after the update.
 
### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N.A

## Release notes
We corrected an update that was meant to enable embed-consent for all users on existing platforms by default. This means that on an existing platform with social_embed module enabled, once the site manager enables the feature at `/admin/config/opensocial/embed-consent`, it will enable opt-in embed-consent option for all users automatically.

## Change Record
N.A

## Translations
N.A

### CODE REVIEW

**Review by Ressinel — COMMENTED** [review]
It works. The setting is enabled for each existing user.

File: modules/social_features/social_embed/social_embed.post_update.php [hunk 1]
```diff
@@ -8,16 +8,22 @@
 use Drupal\Core\Site\Settings;
 
 /**
- * Populate the default value in new field for already existing users.
+ * Empty post update hook.
  */
 function social_embed_post_update_11001_populate_field_embed_content_settings(array &$sandbox): void {
+  // Moved to 11002.
```
Comment (Ressinel) [inline modules/social_features/social_embed/social_embed.post_update.php:14]: @navneet0693  I think we can't move this functionality to a new post update. There can be sites that already have embed `field_user_embed_content_consent` and some users have already could change that setting. So, if we will move this to a new post update all changed settings will be rewritten.

Let me know if I am mistaken.

File: modules/social_features/social_embed/social_embed.post_update.php [hunk 2]
```diff
@@ -77,6 +83,8 @@ function social_embed_post_update_11001_populate_field_embed_content_settings(ar
         )
         ->execute();
       $sandbox['progress']++;
+      \Drupal::messenger()
+        ->addMessage($uid . ' user processed.');
```
[the remark this answers is not in the document]
Comment (navneet0693) [reply modules/social_features/social_embed/social_embed.post_update.php:87]: I will remove the message.

File: modules/social_features/social_embed/social_embed.post_update.php (hunk 1 above)
Comment (navneet0693) [reply modules/social_features/social_embed/social_embed.post_update.php:14]: You are right, I will refactor the code to consider that situation.

Comment (navneet0693) [reply modules/social_features/social_embed/social_embed.post_update.php:14]: @Ressinel Please review the latest changes.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2913-iduqwzqgtittjmxp2ibd3fapd1njpevw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/62712f3c2e91387b17b017ac

**Comment by navneet0693** [comment]
🍒 - picked to 11.3.x, 11.2.x, 11.1.x and 11.0.x

**Pull request merged**